### PR TITLE
Added code to check for valid SNS MessageStructure attributes and con…

### PR DESCRIPTION
**Pull request by jglover**
Added code to check for valid SNS MessageStructure attributes and convert their types (atom to string representation), and tests for SMS and MessageStructure values in SNS requests.

### CODE REVIEW

**Comment by benwilson512** [comment]
The tests however are always welcome :)

File: test/lib/ex_aws/sns_test.exs [hunk 1]
```diff
@@ -51,12 +51,58 @@ defmodule ExAws.SNSTest do
...
+    }
+
+    message_attributes = [
+      %{name: "AWS.SNS.SMS.SenderID", data_type: :string, value: {:string, "sender"}},
+      %{name: "AWS.SNS.SMS.MaxPrice", data_type: :string, value: {:string, "0.8"}},
+      %{name: "AWS.SNS.SMS.SMSType", data_type: :string, value: {:string, "Transactional"}}
+    ]
+    publish_opts = [
+      message_attributes: message_attributes,
+      phone_number: "+15005550006"
```
Comment (benwilson512) [inline test/lib/ex_aws/sns_test.exs]: Do make sure that this isn't a valid phone number.

**Comment by jglover** [comment]
Yeah I did wonder that myself, it also seems like a hassle to maintain rather than let AWS handle it as it will. I'll close this one and open a new one specific to the tests :)

**Comment by benwilson512** [comment]
Thanks!

File: test/lib/ex_aws/sns_test.exs (hunk 1 above)
Comment (jglover) [inline test/lib/ex_aws/sns_test.exs]: It was harder to find a test number than you'd expect it to be ha. 
These are Twilio test credentials, they're an underlying provider for AWS. Not documented via AWS but seems to work just fine, and more importantly isn't going to spam anyone :)
https://www.twilio.com/docs/api/rest/test-credentials

**Pull request closed without merging**